### PR TITLE
cleanup: remove thick provisioning code

### LINKED ISSUE #2795: Remove thickprovision deprecated code from devel branch
As we have already deprecated the thick provisioning in https://github.com/ceph/ceph-csi/pull/2589 and it's already part of 3.5.0 we can plan to remove the support in 3.6.0. This issue is to track the same. 

@nixpanic couple of questions
*  Is it a good time to remove this from the devel branch or should we wait?
* What are the operations we need to support for already created images
       Mounting/Unmount should be enough?

### PR DESCRIPTION
**Pull request by Madhu-1**
This commit removes the thick provisioning code as thick provisioning is deprecated in cephcsi 3.5.0.

fixes: #2795

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: e2e/rbd_helper.go [hunk 1]
```diff
@@ -630,64 +591,6 @@ func deleteBackingRBDImage(f *framework.Framework, pvc *v1.PersistentVolumeClaim
 	return err
 }
 
-// rbdDuImage contains the disk-usage statistics of an RBD image.
```
Comment (Rakshith-R) [inline e2e/rbd_helper.go:633]: Can we keep these helpers around, It'll be needed when writing e2e tests for reclaimspace ?

Comment (Madhu-1) [reply e2e/rbd_helper.go:633]: sure. let me know which functions you want to keep for reclaimspace?

Comment (Rakshith-R) [reply e2e/rbd_helper.go:633]: line 633 to 690.
thanks !

Comment (Madhu-1) [reply e2e/rbd_helper.go:633]: done

**Review by Rakshith-R — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Comment by Madhu-1** [comment]
@mergifyio refresh

**Pull request merged**